src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -289,7 +290,13 @@ class JesBackend extends Backend with LazyLogging {
       val status = run.waitUntilComplete(initializedStatus)
 
       val outputMappings = backendCall.call.task.outputs map {taskOutput =>
-        taskOutput.name -> taskOutput.expression.evaluate(backendCall.lookupFunction, backendCall.engineFunctions)
+        val attemptedValue = taskOutput.expression.evaluate(backendCall.lookupFunction, backendCall.engineFunctions) map { value =>
+          if (taskOutput.wdlType == WdlFileType && WdlFileType.isCoerceableFrom(value.wdlType))
+            jesOutputs find { _.name == value.valueString } map { j => WdlFile(j.gcs) } getOrElse value
+          else
+            value
+        }
+        taskOutput.name -> attemptedValue
```

Comment:
This block of code fixes the core bug... this will evaluate the task output expression.  If it evaluates to a File, find the GCS path and return that, otherwise just return the evaluated value.